Userland/DevTools/HackStudio/Dialogs/NewProjectDialog.cpp

```diff
@@ -93,6 +94,7 @@ NewProjectDialog::NewProjectDialog(GUI::Window* parent)
         if (path.has_value())
             m_create_in_input->set_text(path.value().view());
     };
+    verify_cast<HackStudioWidget>(parent->main_widget())->m_new_project_dialog = RefPtr<NewProjectDialog>(this);
```

Comment:
This is not a good way to attach this data to the main widget.

In HackStudioWidget.cpp, we have this:

```
NonnullRefPtr<GUI::Action> HackStudioWidget::create_new_project_action()
{
    return GUI::Action::create("&Project...", Gfx::Bitmap::try_load_from_file("/res/icons/16x16/hackstudio-project.png"sv).release_value_but_fixme_should_propagate_errors(), [this](const GUI::Action&) {
        auto dialog = NewProjectDialog::construct(window());
        dialog->set_icon(window()->icon());
        auto result = dialog->exec();

        if (result == GUI::Dialog::ExecResult::OK && dialog->created_project_path().has_value())
            open_project(dialog->created_project_path().value());
    });
}
```
Just assign `this->m_new_project_dialog = dialog;` in this function, if it's actually correct to save it.

Reply:
The action can simply warn on unsaved changes before constructing the dialog. This is the standard workflow